File: api-surface/src/services/api-security/securityController.ts
// @ts-ignore
/* eslint-disable */
import {request} from '@umijs/max';

/** 此处后端没有提供注释 POST /api/admin/page */
export async function viewApiAdminPage(
  body: API.ApiAdminPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultApiAdminPageVO>(`/security/api/admin/page`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /api/creator/${param0} */
export async function viewApiCreator(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.viewApiCreatorParams,
  options?: { [key: string]: any },
) {
  const { accountId: param0, ...queryParams } = params;
  return request<API.ResultApiCreatorVO>(`/security/api/creator/${param0}`, {
    method: 'POST',
    params: { ...queryParams },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /csrf/token */
export async function getCsrfToken(options?: { [key: string]: any }) {
  return request<API.ResultVoid>(`/security/csrf/token`, {
    method: 'GET',
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /request/user/info/result */
export async function getClientUserInfoResult(
  body: API.ClientUserInfoQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultClientUserInfoDTO>(`/security/request/user/info/result`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/admin/page */
export async function viewUserAdminPage(
  body: API.UserAdminPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultUserAdminPageVO>(`/security/user/admin/page`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/api/digest/page */
export async function viewUserApiDigestPage(
  body: API.UserApiDigestPageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultUserApiDigestPageVO>(`/security/user/api/digest/page`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /user/api/format/and/quantity/usage */
export async function viewUserApiFormatAndQuantityUsage(
  body: API.UserApiFormatAndQuantityUsageQuery,
  options?: { [key: string]: any },
) {
  return request<API.ResultUserApiFormatAndQuantityUsageVO>(
    `/security/user/api/format/and/quantity/usage`,
    {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      data: body,
      ...(options || {}),
    },
  );
}

/** 此处后端没有提供注释 POST /user/api/test */
export async function testUserApi(body: API.UserApiTestDTO, options?: { [key: string]: any }) {
  return request<API.ResultUserApiTestVO>(`/security/user/api/test`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /verification/code */
export async function getVerificationCode(
  body: API.VerificationCodeSendingDTO,
  options?: { [key: string]: any },
) {
  return request<API.ResultVoid>(`/security/verification/code`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: api-security/src/main/java/pers/project/api/security/controller/SecurityController.java
package pers.project.api.security.controller;

import jakarta.servlet.http.HttpServletRequest;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;
import pers.project.api.common.model.Result;
import pers.project.api.common.model.dto.ClientUserInfoDTO;
import pers.project.api.common.model.query.*;
import pers.project.api.common.model.vo.ApiAdminPageVO;
import pers.project.api.common.model.vo.UserAdminPageVO;
import pers.project.api.common.model.vo.UserApiDigestPageVO;
import pers.project.api.common.model.vo.UserApiFormatAndQuantityUsageVO;
import pers.project.api.common.util.ResultUtils;
import pers.project.api.common.validation.constraint.SnowflakeId;
import pers.project.api.security.model.dto.UserApiTestDTO;
import pers.project.api.security.model.dto.VerificationCodeSendingDTO;
import pers.project.api.security.model.vo.ApiCreatorVO;
import pers.project.api.security.model.vo.UserApiTestVO;
import pers.project.api.security.service.SecurityService;

/**
 * Security 项目控制器
 *
 * @author Luo Fei
 * @date 2023/03/24
 */
@Slf4j
@Validated
@RestController
@RequiredArgsConstructor
public class SecurityController {

    private final SecurityService securityService;

    // region OpenFeign
    @PostMapping("/request/user/info/result")
    public Result<ClientUserInfoDTO> getClientUserInfoResult(@Valid @RequestBody ClientUserInfoQuery userInfoQuery) {
        ClientUserInfoDTO clientUserInfoDTO = securityService.getClientUserInfoDTO(userInfoQuery);
        return ResultUtils.success(clientUserInfoDTO);
    }
    // endregion

    @GetMapping("/csrf/token")
    public Result<Void> getCsrfToken(HttpServletRequest request) {
        securityService.loadDeferredCsrfToken(request);
        return ResultUtils.success();
    }

    @PostMapping("/verification/code")
    public Result<Void> getVerificationCode(@Valid @RequestBody VerificationCodeSendingDTO codeSendingDTO) {
        securityService.sendVerificationCode(codeSendingDTO);
        return ResultUtils.success();
    }

    @PostMapping("/user/api/digest/page")
    public Result<UserApiDigestPageVO> viewUserApiDigestPage
            (@Valid @RequestBody UserApiDigestPageQuery pageQuery) {
        UserApiDigestPageVO pageVO = securityService.getUserApiDigestPageVO(pageQuery);
        return ResultUtils.success(pageVO);
    }

    @PostMapping("/user/api/format/and/quantity/usage")
    public Result<UserApiFormatAndQuantityUsageVO> viewUserApiFormatAndQuantityUsage
            (@Valid @RequestBody UserApiFormatAndQuantityUsageQuery query) {
        UserApiFormatAndQuantityUsageVO formatAndQuantityUsageVO
                = securityService.getUserApiFormatAndQuantityUsageVO(query);
        return ResultUtils.success(formatAndQuantityUsageVO);
    }

    @PostMapping("/user/admin/page")
    public Result<UserAdminPageVO> viewUserAdminPage
            (@Valid @RequestBody UserAdminPageQuery pageQuery) {
        UserAdminPageVO pageVO = securityService.getUserAdminPageVO(pageQuery);
        return ResultUtils.success(pageVO);
    }

    @PostMapping("/api/admin/page")
    public Result<ApiAdminPageVO> viewApiAdminPage
            (@Valid @RequestBody ApiAdminPageQuery pageQuery) {
        ApiAdminPageVO pageVO = securityService.getApiAdminPageVO(pageQuery);
        return ResultUtils.success(pageVO);
    }

    @PostMapping("/api/creator/{accountId}")
    public Result<ApiCreatorVO> viewApiCreator
            (@SnowflakeId @PathVariable String accountId) {
        ApiCreatorVO apiCreatorVO = securityService.getApiCreatorVO(accountId);
        return ResultUtils.success(apiCreatorVO);
    }

    @PostMapping("/user/api/test")
    public Result<UserApiTestVO> testUserApi
            (@Valid @RequestBody UserApiTestDTO UserApiTestDTO) {
        UserApiTestVO userApiTestVO = securityService.getUserApiTestVO(UserApiTestDTO);
        return ResultUtils.success(userApiTestVO);
    }

}
